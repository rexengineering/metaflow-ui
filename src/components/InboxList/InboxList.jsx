import {
  IconButton,
  List,
  ListItem,
  makeStyles,
  Typography,
} from "@material-ui/core";
import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import PropTypes from "prop-types";
import fontAwesomeIcon from "../../utils/propsValidators";

const useStyles = makeStyles((theme) => ({
  listItem: {
    display: "flex",
    alignItems: "center",
    justifyContent: "space-around",
    padding: theme.spacing(2),
    width: "100%",
    color: theme.palette.common.white,
  },
}));

function InboxList({ messagesItems, onIboxItemSelected }) {
  const styles = useStyles();
  return (
    <List>
      {Array.isArray(messagesItems) &&
        messagesItems.map(({ icon, cellphoneNumber, color, time }) => (
          <ListItem button key={cellphoneNumber} onClick={onIboxItemSelected}>
            <IconButton style={{ background: color }}>
              <FontAwesomeIcon icon={icon} />
            </IconButton>
            <div className={styles.listItem}>
              <Typography variant="body1">{cellphoneNumber}</Typography>
              <Typography variant="body1">{time}</Typography>
            </div>
          </ListItem>
        ))}
    </List>
  );
}

InboxList.propTypes = {
  messagesItems: PropTypes.arrayOf(
    PropTypes.shape({
      icon: fontAwesomeIcon.isRequired,
      color: PropTypes.string.isRequired,
      hasShadow: PropTypes.bool.isRequired,
      time: PropTypes.string.isRequired,
      cellphoneNumber: PropTypes.string.isRequired,
    })
  ).isRequired,
  onIboxItemSelected: PropTypes.func.isRequired,
};

export default InboxList;
