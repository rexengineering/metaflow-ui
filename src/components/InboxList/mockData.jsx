import { faCommentAltLines } from "@fortawesome/pro-light-svg-icons/faCommentAltLines";
import { faPhoneAlt } from "@fortawesome/pro-light-svg-icons/faPhoneAlt";
import { faCommentAlt } from "@fortawesome/pro-regular-svg-icons/faCommentAlt";

const data = [
  {
    icon: faCommentAlt,
    color: "#FF685F",
    hasShadow: true,
    time: "01:26",
    cellphoneNumber: "(310) - 555 - 5855",
  },
  {
    icon: faCommentAltLines,
    color: "#89C5CC",
    hasShadow: true,
    time: "13:55",
    cellphoneNumber: "(310) - 555 - 5523",
  },
  {
    icon: faPhoneAlt,
    color: "#BDBDBD",
    hasShadow: true,
    time: "14:02",
    cellphoneNumber: "(310) - 555 - 5555",
  },
];

export default data;
