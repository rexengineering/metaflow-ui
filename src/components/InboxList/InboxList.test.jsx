import { render, screen, fireEvent } from "@testing-library/react";
import React from "react";

import InboxList from "./InboxList";
import messageItems from "./mockData";

describe("<InboxList />", () => {
  const renderInboxItems = (onIboxItemSelected = () => {}) => (
    <InboxList
      onIboxItemSelected={onIboxItemSelected}
      messagesItems={messageItems}
    />
  );

  it("should render", () => {
    const { asFragment } = render(renderInboxItems());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(renderInboxItems());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });

  it("should render all passed inbox items", () => {
    render(renderInboxItems());
    messageItems.forEach(({ cellphoneNumber }) =>
      expect(screen.getByText(cellphoneNumber)).toBeInTheDocument()
    );
  });

  it("should handle all click events properly", () => {
    const onIboxItemSelected = jest.fn();
    render(renderInboxItems(onIboxItemSelected));

    const buttons = screen.getAllByRole("button");
    buttons.forEach((button) => {
      fireEvent.click(button);
    });
    expect(onIboxItemSelected).toHaveBeenCalledTimes(buttons.length);
  });
});
