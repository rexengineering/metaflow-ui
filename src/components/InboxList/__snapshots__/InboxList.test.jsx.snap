// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InboxList /> should match snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        style="background: rgb(255, 104, 95);"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-comment-alt fa-w-16 "
            data-icon="comment-alt"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 0H64C28.7 0 0 28.7 0 64v288c0 35.3 28.7 64 64 64h96v84c0 7.1 5.8 12 12 12 2.4 0 4.9-.7 7.1-2.4L304 416h144c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64zm16 352c0 8.8-7.2 16-16 16H288l-12.8 9.6L208 428v-60H64c-8.8 0-16-7.2-16-16V64c0-8.8 7.2-16 16-16h384c8.8 0 16 7.2 16 16v288z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="makeStyles-listItem-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          (310) - 555 - 5855
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          01:26
        </p>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        style="background: rgb(137, 197, 204);"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-comment-alt-lines fa-w-16 "
            data-icon="comment-alt-lines"
            data-prefix="fal"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 0H64C28.7 0 0 28.7 0 64v288c0 35.3 28.7 64 64 64h96v84c0 7.1 5.8 12 12 12 2.4 0 4.9-.7 7.1-2.4L304 416h144c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64zm32 352c0 17.6-14.4 32-32 32H293.3l-8.5 6.4L192 460v-76H64c-17.6 0-32-14.4-32-32V64c0-17.6 14.4-32 32-32h384c17.6 0 32 14.4 32 32v288zM280 240H136c-4.4 0-8 3.6-8 8v16c0 4.4 3.6 8 8 8h144c4.4 0 8-3.6 8-8v-16c0-4.4-3.6-8-8-8zm96-96H136c-4.4 0-8 3.6-8 8v16c0 4.4 3.6 8 8 8h240c4.4 0 8-3.6 8-8v-16c0-4.4-3.6-8-8-8z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="makeStyles-listItem-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          (310) - 555 - 5523
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          13:55
        </p>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        style="background: rgb(189, 189, 189);"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-phone-alt fa-w-16 "
            data-icon="phone-alt"
            data-prefix="fal"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M493.09 351.3L384.7 304.8a31.36 31.36 0 0 0-36.5 8.9l-44.1 53.9A350 350 0 0 1 144.5 208l53.9-44.1a31.35 31.35 0 0 0 8.9-36.49l-46.5-108.5A31.33 31.33 0 0 0 125 .81L24.2 24.11A31.05 31.05 0 0 0 0 54.51C0 307.8 205.3 512 457.49 512A31.23 31.23 0 0 0 488 487.7L511.19 387a31.21 31.21 0 0 0-18.1-35.7zM456.89 480C222.4 479.7 32.3 289.7 32.1 55.21l99.6-23 46 107.39-72.8 59.5C153.3 302.3 209.4 358.6 313 407.2l59.5-72.8 107.39 46z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="makeStyles-listItem-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          (310) - 555 - 5555
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          14:02
        </p>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </ul>
</DocumentFragment>
`;
